cmd/pilot-agent/main.go

```diff
@@ -118,24 +136,18 @@ var (
 )
 
 func init() {
+	proxyCmd.PersistentFlags().StringVar(&config, "config", "",
```

Comment:
Can you update description to indicate the value of `config` is the YAML/JSON content itself and not a filename? It wasn't obvious to me until I reviewed the initializer code that uses it.

Reply:
Yes, will fix.

Reply:
renamed to `configInline`

Reply:
My point was that passing a yaml file as an 'inline' command line parameter is not how 
either CLI or yaml were meant to be used.

Reply:
Please suggest a way without enumerating all proto fields and creating flags for each.

Reply:
Any better if we pass JSON? That's a common pattern with other command line tools, e.g. 

```bash
curl -H "Content-Type: application/json" -X POST -d '{"username":"xyz","password":"xyz"}' http://localhost:3000/api/login
```

Reply:
And as a general comment - using a proto and yaml to pass configuration seems super complex given that in practice we have a dozen or so settings that could be cleanly expressed as flags or env ( both supported by cobra ). 

Just because someday we may have nested config doesn't require both proto and yaml - the flags have self-included doc and defaults, are well understood, etc. 

I'm not aware of any benefit of having the proto in the api directory that justifies the added compexity.

Reply:
YAML is a superset of JSON. Flags are in-code, proto is a declarative spec. We generated doc pages from the proto, can't be done that easily for flags. Constraints are also easier to express over proto than flags. Anyways, since we do processing of the values, it's better to treat them as first class API objects, rather than a bunch of strings in some flags.

Reply:
I'm not sure what is the problem with having one field/flag for each field - it's a dozen or so - 
having them in the .proto or in the code is pretty much the same amount of code. Yaml may be
best for complex configs ( not our case ), proto is great for network transport - but it seems in
this case flags would be a much better user experience. Even k8s allow pretty much all operations to be done using flags.

Reply:
Kubernetes is also trying to switch to configmaps whenever possible to let them change dynamically. We're in a better position here with not having a bazillion flags.